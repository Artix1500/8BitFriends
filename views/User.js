import React from 'react';
import {View, StyleSheet, Button, Text} from 'react-native';
import Header from '../components/Header';
import {Avatar} from 'react-native-elements';

const Login = ({navigation, route}) => {
  return (
    <View style={styles.container}>
      <Header title="8BitFriends" />

      <View style={styles.userView}>
        <Avatar
          rounded
          size="xlarge"
          title={route.params.user.login}
          source={{
            uri: route.params.user.avatarUrl,
          }}
        />
        <Text style={styles.smallFont}>login:</Text>
        <Text style={styles.bigFont}>{route.params.user.login}</Text>
        <Text style={styles.smallFont}>name:</Text>
        <Text style={styles.bigFont}>{route.params.user.name}</Text>
        <Text style={styles.smallFont}>location:</Text>
        <Text style={styles.bigFont}>{route.params.user.location}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  userView: {
    paddingTop: 30,
    alignItems: 'center',
  },
  smallFont: {
    marginTop: 20,
    fontSize: 12,
    color: 'grey',
  },
  bigFont: {
    fontSize: 20,
  },
});

export default Login;
