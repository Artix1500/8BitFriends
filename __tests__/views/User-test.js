/**
 * @format
 */

import 'react-native';
import React from 'react';
import User from '../../views/User';

// Note: test renderer must be required after react-native.
import render from 'react-test-renderer';

describe('User view', () => {
  it('Search value is empty after mount', () => {
    const data = {
      __typename: 'User',
      name: 'Artur Haczek',
      login: 'Artix1500',
      avatarUrl: 'https://avatars2.githubusercontent.com/u/38567097?v=4',
      location: null,
      repositories: {totalCount: 12},
    };

    const tree = render
      .create(<User route={{params: {user: data}}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
