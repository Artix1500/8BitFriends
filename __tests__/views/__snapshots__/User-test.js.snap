// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User view Search value is empty after mount 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "indigo",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 23,
          "textAlign": "center",
        }
      }
    >
      8BitFriends
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 75,
          "height": 150,
          "width": 150,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "#bdbdbd",
            "borderRadius": 75,
            "flex": 1,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <Image
          onLoad={[Function]}
          source={
            Object {
              "uri": "https://avatars2.githubusercontent.com/u/38567097?v=4",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": null,
                "width": null,
              },
            ]
          }
          testID="RNE__Image"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        />
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "flex": 1,
                "height": null,
                "justifyContent": "center",
                "width": null,
              }
            }
            testID="RNE__Image__placeholder"
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontSize": 75,
                  "textAlign": "center",
                }
              }
            >
              Artix1500
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "height": null,
              "width": null,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 12,
          "marginTop": 20,
        }
      }
    >
      login:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Artix1500
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 12,
          "marginTop": 20,
        }
      }
    >
      name:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Artur Haczek
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 12,
          "marginTop": 20,
        }
      }
    >
      location:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;
